src/compiler/checker.ts

```diff
@@ -30243,10 +30243,11 @@ namespace ts {
         }
 
         function getTypeArgumentsFromNodes(typeArgumentNodes: readonly TypeNode[], typeParameters: readonly TypeParameter[], isJs: boolean): readonly Type[] {
-            const typeArguments = typeArgumentNodes.map(getTypeOfNode);
+            let typeArguments = typeArgumentNodes.map(getTypeOfNode);
             while (typeArguments.length > typeParameters.length) {
                 typeArguments.pop();
             }
+            typeArguments = fillMissingTypeArguments(typeArguments, typeParameters, getMinTypeArgumentCount(typeParameters), isJs);
             while (typeArguments.length < typeParameters.length) {

```

Comment:
In what cases does this loop still run?

Reply:
like this(the example is from `instantiateGenericClassWithWrongNumberOfTypeArguments.ts`):
```
class D<T, U> {
    x: T
    y: U
}

var d = new D<number>();
// when hover on `d`, it shows `var d: D<number, unknown>`
//                                               ^^^^^^ here is added by this loop, and if `U` has a constraint, it would be the constraint.
```
`fillMissingTypeArguments` only fills `typeArgument` when there are valid default type parameter. It would not add any other default values.

Reply:
And, we would better to mention this case: 
```
interface A { a: number }
interface B extends A { b: string }

class D<T, U, V extends A = B> {
    x: T
    y: U
    z: V
}

var d = new D<number>();
// when hover on `d`, it shows `var d: D<number, unknown, A>`
// `A` is its constraint rather than the default type parameter `B`, this PR does not change this behavior now
```
However, it is a little different: there is an error on the number of type arguments. So I feel that both methods are OK? Or the default type parameter is better?

Reply:
Hm, I feel like using the default here makes a little more sense, but I don’t know that it’s super critical since it’s an error anyway.

Reply:
Now prefer default type parameter than constraint for quick info. Correspond tests are added.

Reply:
It should be noted that this change doesn’t just affect quick info; it actually affects the type instantiated in these kinds of error scenarios. I’m surprised there weren’t any existing baselines that needed updating. In your example above, you can check the type of `d.z` and see that in main, the type is `A`, and this branch, the type is `B`.

Reply:
The new behavior is more intuitive to me, but others should weigh in. The most conservative thing would be to leave it as it was, against my earlier advice.

Reply:
Thanks for pointing it out, I did not notice it! This does make sense. And, I am surprised too :)
But do we really need to revert to last commit? For:
1. we are doing things in the right way
2. anyway, it is just an error condition which finally need be fixed by user

Again, I feel both methods are OK. The final decision is up to you~